### PR TITLE
qemu_guest_agent: verify the blacklist of guest agent

### PR DESCRIPTION
**Pull request by xiagao**
ID: 1776118
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
(1/2) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_check_blacklist: PASS (69.68 s)
 (2/2) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_check_blacklist: PASS (125.62 s)

**Comment by xiagao** [comment]
@xiangchunfu call for your review,thanks.

**Comment by xiangchunfu** [comment]
@xiagao  In order to ensure blacklist is working.  Do you need add a new agent command to black list?  I found you only remove agent command from black list.

**Comment by xiangchunfu** [comment]
@xiagao ignore my comments above.  It cover both(add and remove agent command)

Acked by: Xiangchun Fu xfu@redhat.com

**Comment by xiagao** [comment]
@lolyu  Could you help review this patch? Thanks.

**Comment by xiagao** [comment]
@vivianQizhu Could you help review this patch? If you are busy,it's ok,I will ask others to review.
Thanks.

**Comment by xiagao** [comment]
@lolyu  don't forget this patch when you are available, thanks.

File: qemu/tests/qemu_guest_agent.py [hunk 1]
```diff
@@ -2279,6 +2279,61 @@ def gagent_check_user_logoff(self, test, params, env):
...
+        """
+        session = self._get_session(params, None)
+        self._open_session_list.append(session)
+
+        error_context.context("Try to execute guest-file-open command which"
+                              " is in blacklist by default.", logging.info)
+
+        ranstr = utils_misc.generate_random_string(5)
+        guest_file = "/tmp/qgatest" + ranstr
+        try:
```
Comment (lolyu) [inline qemu/tests/qemu_guest_agent.py]: better to put this large try block into a function.

Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py]: large try block?
or you mean make a function for the two try block operation?

**Comment by xiagao** [comment]
Updated.
 (1/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_check_blacklist: PASS (70.38 s)
 (2/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_check_blacklist: PASS (78.42 s)

**Comment by xiagao** [comment]
@lolyu don't forget this patch when you are available, thanks.

File: qemu/tests/qemu_guest_agent.py [hunk 2]
```diff
@@ -2279,6 +2279,67 @@ def gagent_check_user_logoff(self, test, params, env):
...
+                test.fail("%s cmd is not in blacklist,"
+                          " pls have a check." % qga_cmd)
+
+        session = self._get_session(params, None)
+        self._open_session_list.append(session)
+
+        error_context.context("Try to execute guest-file-open command which"
+                              " is in blacklist by default.", logging.info)
+
+        ranstr = utils_misc.generate_random_string(5)
```
Comment (lolyu) [inline qemu/tests/qemu_guest_agent.py]: `randstr` will be better

File: qemu/tests/qemu_guest_agent.py [hunk 3]
```diff
@@ -2279,6 +2279,67 @@ def gagent_check_user_logoff(self, test, params, env):
...
+            """
+            check if qga cmd is disabled.
+            """
+            try:
+                if qga_cmd == "guest-file-open":
+                    self.gagent.guest_file_open(guest_file, mode="a+")
+                else:
+                    self.gagent.cmd(qga_cmd)
+            except guest_agent.VAgentCmdError as detail:
+                if re.search("has been disabled", str(detail)):
```
Comment (lolyu) [inline qemu/tests/qemu_guest_agent.py]: why not search to find the keyword `qga_cmd`?

File: qemu/tests/qemu_guest_agent.py (hunk 2 above)
Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py]: ok, will update

File: qemu/tests/qemu_guest_agent.py (hunk 3 above)
Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py]: you mean add cmd like guest-file-open ? Actually it works if not search qga_cmd, but add the cmd will be better, will update.
{"error": {"class": "GenericError", "desc": "The command guest-file-open has been disabled for this instance"}}

**Comment by xiagao** [comment]
@lolyu Update patch according to your suggestion. Pls have a check, thanks.

File: qemu/tests/qemu_guest_agent.py [hunk 4]
```diff
@@ -2279,6 +2279,67 @@ def gagent_check_user_logoff(self, test, params, env):
...
+        def bl_check(qga_cmd):
+            """
+            check if qga cmd is disabled.
+            """
+            try:
+                if qga_cmd == "guest-file-open":
+                    self.gagent.guest_file_open(guest_file, mode="a+")
+                else:
+                    self.gagent.cmd(qga_cmd)
+            except guest_agent.VAgentCmdError as detail:
```
Comment (lolyu) [inline qemu/tests/qemu_guest_agent.py]: Line 2301 and 2302 is incorrect.

File: qemu/tests/qemu_guest_agent.py [hunk 5]
```diff
@@ -215,7 +215,7 @@ def gagent_verify(self, params, vm):
         if not self.gagent:
             self.test.error("Could not find guest agent object "
                             "for VM '%s'" % vm.name)
-
+        time.sleep(10)
```
Comment (lolyu) [inline qemu/tests/qemu_guest_agent.py]: why sleep here? better to use `wait_for`.

Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py]: the sleep is useless. delete it.

File: qemu/tests/qemu_guest_agent.py (hunk 4 above)
Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py]: modified.

**Comment by xiagao** [comment]
(1/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_check_blacklist: PASS (77.67 s)
 (2/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_check_blacklist: PASS (71.48 s)

**Pull request merged**